pytest_fixtures/certificate.py

```diff
@@ -0,0 +1,150 @@
+from pathlib import Path
+
+import pytest
+from fauxfactory import gen_string
+
+from robottelo import ssh
+from robottelo.config import settings
+from robottelo.helpers import get_data_file
+from robottelo.ssh import get_connection
+from robottelo.ssh import upload_file
+
+
+@pytest.fixture(scope="module")
+def cert_data():
+    """Get host name, scripts, and create working directory."""
+
+    def _cert_data(hostname=None):
+        sat6_hostname = hostname or settings.server.hostname
+        capsule_hostname = 'capsule.example.com'
+        key_file_name = '{0}/{0}.key'.format(sat6_hostname)
+        cert_file_name = '{0}/{0}.crt'.format(sat6_hostname)
+        ca_bundle_file_name = 'cacert.crt'
+        success_message = 'Validation succeeded'
+        cert_data = {
+            'sat6_hostname': sat6_hostname,
+            'capsule_hostname': capsule_hostname,
+            'key_file_name': key_file_name,
+            'cert_file_name': cert_file_name,
+            'ca_bundle_file_name': ca_bundle_file_name,
+            'success_message': success_message,
+        }
+        return cert_data
+
+    return _cert_data
+
+
+@pytest.fixture()
+def cert_setup(cert_data):
+    """copy all configuration files to satellite host for generating custom certs"""
+
+    def _cert_setup(hostname=None):
+        # Need a subdirectory under ssl-build with same name as Capsule name
+        hostname = hostname or settings.server.hostname
+        _cert_data = cert_data(hostname)
+        with get_connection(timeout=100, hostname=hostname) as connection:
+            connection.run('mkdir ssl-build/{}'.format(_cert_data['capsule_hostname']))
+            # Ignore creation error, but assert directory exists
+            assert connection.run('test -e ssl-build/{}'.format(_cert_data['capsule_hostname']))
+        upload_file(
+            local_file=get_data_file('generate-ca.sh'),
+            remote_file='generate-ca.sh',
+            hostname=hostname,
+        )
+        upload_file(
+            local_file=get_data_file('generate-crt.sh'),
+            remote_file='generate-crt.sh',
+            hostname=hostname,
+        )
+        upload_file(
+            local_file=get_data_file('openssl.cnf'), remote_file='openssl.cnf', hostname=hostname
+        )
+        # create the CA cert.
+        with get_connection(timeout=300, hostname=hostname) as connection:
+            result = connection.run('echo 100001 > serial')
+            result = connection.run('bash generate-ca.sh')
+            assert result.return_code == 0
+        # create the Satellite's cert
+        with get_connection(timeout=300, hostname=hostname) as connection:
+            result = connection.run(
+                'yes | bash {} {}'.format('generate-crt.sh', _cert_data['sat6_hostname'])
```

Comment:
f-strings for these - one of the arguments is a bare string, which doesn't need to be formatted in and can be part of the string directly.